src/modules/aws_ec2/mappers/instance_block_device_mapping.ts

```diff
@@ -0,0 +1,259 @@
+import { EC2, InstanceBlockDeviceMapping as AWSInstanceBlockDeviceMapping } from '@aws-sdk/client-ec2';
+import { Volume as AWSVolume } from '@aws-sdk/client-ec2';
+import { WaiterState } from '@aws-sdk/util-waiter';
+
+import { AwsEc2Module } from '..';
+import { AWS, crudBuilder } from '../../../services/aws_macros';
+import { Context, Crud, IdFields, MapperBase } from '../../interfaces';
+import { Instance, InstanceBlockDeviceMapping, VolumeState } from '../entity';
+
+export class InstanceBlockDeviceMappingMapper extends MapperBase<InstanceBlockDeviceMapping> {
+  module: AwsEc2Module;
+  entity = InstanceBlockDeviceMapping;
+  generateId = (fields: IdFields) => {
+    const requiredFields = ['instanceId', 'deviceName', 'volumeId', 'region'];
+    if (
+      Object.keys(fields).length !== requiredFields.length &&
+      !Object.keys(fields).every(fk => requiredFields.includes(fk))
+    ) {
+      throw new Error(`Id generation error. Valid fields to generate id are: ${requiredFields.join(', ')}`);
+    }
+    return `${fields.instanceId}|${fields.deviceName}|${fields.volumeId}|${fields.region}`;
+  };
+  entityId = (e: InstanceBlockDeviceMapping) => {
+    return this.module.instanceBlockDeviceMapping.generateId({
+      instanceId: e.instance?.instanceId ?? '',
+      deviceName: e.deviceName,
+      volumeId: e.volume?.volumeId ?? '',
+      region: e.region,
+    });
+  };
```

Comment:
having this here should be overriding your recently added `@cloudId` decorators for this entity. Ideally, we should just keep one of them to avoid confusions.